Clarify that v1.0 of Matrix was a symbolic release (#2045)
matrix-org/matrix-spec: content/_index.md

roughly represents a breaking change, `Y` a backwards-compatible change, and
`Z` a patch or insignificant alteration to the API.

> `v1.0` of Matrix was a symbolic version released on June 10th, 2019 with the

Author: Should we link to the matrix.org blog post of the release?

> `v1.0` of Matrix was a symbolic version released on June 10th, 2019 with the
> following minimum API versions:

Reviewer: I don't really think that "symbolic release" is going to mean much to the unfamiliar reader. Maybe:

```suggestion
The current global versioning system was introduced with `v1.1`. 
[Matrix 1.0][https://matrix.org/blog/2019/06/11/introducing-matrix-1-0-and-the-matrix-org-foundation/]
did not correspond directly to a specification version; instead, it was based on the following
versions for the individual APIs:
```

Author: Applied

| Push Gateway API        | r0.1.0  |

> | Room Version            | v5      |

Reviewer: Separate problem, but I don't really think that Matrix 1.0 corresponded to room v5: if you look at, say, https://github.com/matrix-org/matrix-spec-proposals/blob/f44483a8cf0f174376647ac5a777ad6145b5f195/specification/index.rst#specification-versions, it's clear that *all* room versions from v1 to v5 were part of Matrix 1.0.

So, maybe while we're here:

```suggestion
| Room Versions           | v1-v5   |
```

cc @turt2live, as the original author of this table

Reviewer 2: at a spec level room versions aren't linear, so we'd define this as `1, 2, 3, 4, 5` (yes, v5 is also wrong - it's just `5`).

the backing history here is that it felt weird to list out the room versions like that, so a slight trick was applied: because all the room versions inherit from each other, even though they don't have to, we can simply point to the "latest" which drags the remainder into scope as well. The deeper rationale is that by listing fewer numbers, we made the table less scary. Since we're well past the 1.0 launch window though, we can probably reintroduce a lot of that complexity.

Reviewer: ```suggestion
| Room Versions           | 1, 2, 3, 4, 5 |
```

then

Author: Applied

> It is not used in this specification and should not be returned by servers in
> the [`GET /_matrix/client/versions`](/client-server-api/#get_matrixclientversions)
> endpoint. The first version using the current global versioning system was `v1.1`.

Reviewer: ```suggestion
`v1.0` should **not** be returned by servers in
the [`GET /_matrix/client/versions`](/client-server-api/#get_matrixclientversions)
response.
```

Author: Applied
